task/ab-load/0.1/ab-load.yaml

```diff
@@ -0,0 +1,163 @@
+      type: string
+      description: Seconds to max. wait for each response
+      default: "5"
+    - name: method
+      type: string
+      description: Method name
+      default: "GET"
+    - name: additionnalFlags
+      type: string
+      description: "All ab command supported flag (ex: -k for keepalive)"
+      default: "-k -I"
+    - name: type
+      type: string
+      description: "The http type of payload sended to the target endpoint"
+      default: "application/x-www-form-urlencoded"
+  steps:
+    - image: quay.io/startx/apache:fc35
+      command: ["/bin/bash", "-c"]
+      args: 
+        - |-
+          #!/bin/bash
+
+          # execute the AB benchmark for GET load
+          function abGET {
+            local url=${1:-myurl}
+            local size=${2:-1000}
+            local concurrency=${3:-10}
+            local timeout=${4:-10}
+            local additionnal=${5}
+            echo "= Start http test GET ${AB_URL}"
+            echo "  - ${size} requests"
+            echo "  - ${concurrency} concurrent jobs"
+            ab -m GET -n "$size" -c "$concurrency" -s "$timeout" $additionnal "$url"
+          }
+
+          # execute the AB benchmark for POST load
+          function abPOST {
+            local file=${AB_FILE}
+            local type=${AB_TYPE:-application/x-www-form-urlencoded}
+            local url=${1:-myurl}
+            local size=${2:-1000}
+            local concurrency=${3:-10}
+            local timeout=${4:-10}
+            local additionnal=${5}
+            echo "= Start http test POST ${AB_URL}"
+            echo "  - ${size} requests"
+            echo "  - ${concurrency} concurrent jobs"
+            ab -n "$size" \
+               -c "$concurrency" \
+               -s "$timeout" \
+               -p "$file" \
+               -T '${type}; charset=UTF-8' \
+            $additionnal "$url"
+          }
+
+          # execute the AB benchmark for PUT load
+          function abPUT {
+            local file=${AB_FILE}
+            local type=${AB_TYPE:-application/x-www-form-urlencoded}
+            local url=${1:-myurl}
+            local size=${2:-1000}
+            local concurrency=${3:-10}
+            local timeout=${4:-10}
+            local additionnal=${5}
+            echo "= Start http test PUT ${AB_URL}"
+            echo "  - ${size} requests"
+            echo "  - ${concurrency} concurrent jobs"
+            ab -n "$size" \
+               -c "$concurrency" \
+               -s "$timeout" \
+               -u "$file" \
+               -T '${type}; charset=UTF-8' \
+            $additionnal "$url"
+          }
+
+          # execute the AB benchmark for DELETE load
+          function abDELETE {
+            local url=${1:-myurl}
+            local size=${2:-1000}
+            local concurrency=${3:-10}
+            local timeout=${4:-10}
+            local additionnal=${5}
+            echo "= Start http test DELETE ${AB_URL}"
+            echo "  - ${size} requests"
```

Comment:
can we merge abGET and abDELETE functions as both look similar with just the difference of GET and DELETE?

Reply:
corrected into https://github.com/tektoncd/catalog/pull/1002/commits/a7b52b5664e0206f5aba47916c6e87bdd4f9a5ce